### PR TITLE
Implemented true 2D rendering

### PR DESCRIPTION
**Pull request by Raelr**
### Description

This PR adds a new 2D renderer which renders 'true' 2D. This 2D implemention starts from the top left of the screen as (0,0), and all positional elements positioned via their top left coordinates. The PR adds the following:

1) A new UI Camera which renders UI in front of the screen
2) A new Renderer2D type which renders quads and font characters to 2D space
3) Fixes an old bug with text rendering where uv coordinates were not properly being calculated.
4) Uses SDF fonts to sharpen text and remove text blur

### The PR has been...
- [x] provided a reasonable name that is not just the branch name (e.g "Added Vulkan render delegate")
- [x] linked to its related issue
- [x] assigned a reviewer from the team
- [x] labelled appropriately

### The code has been...
- [x] made mergable and free of conflicts in relation to `master` *(according to GitHub)*
- [x] tested in a packaged state using the `package` targets
- [x] pulled to the reviewer's machine and reasonably tested

### CODE REVIEW

File: examples/render/src/main.cpp [hunk 1]
```diff
@@ -148,18 +152,26 @@ int main()
 
         float aspect = renderer.GetAspectRatio();
 
+        float panelWidth = window.GetWidth();
+        float panelHeight = 50.f;
+
         camera.UpdatePerspectiveProjection(Siege::Float::Radians(50.f), aspect, 0.1f, 100.f);
 
+        camera2
+            .UpdateOthrographicProjection(0, window.GetWidth(), 0, window.GetHeight(), 0.1f, 100.f);
```
Comment (J-Mo63) [inline examples/render/src/main.cpp:161]:
```suggestion
            .UpdateOrthographicProjection(0, window.GetWidth(), 0, window.GetHeight(), 0.1f, 100.f);
```

**Review by J-Mo63 — APPROVED** [review]
Looking good. I'll push up some quick changes with my comments applied and it should be good to commit.

File: examples/render/src/main.cpp (hunk 1 above)
Comment (Raelr) [reply examples/render/src/main.cpp:161]: Done

**Comment by J-Mo63** [comment]
@Raelr should be good to go whenever suits.

File: engine/render/renderer/renderer/Renderer2D.cpp [hunk 2]
```diff
@@ -8,110 +8,283 @@
...
+    for (int i = MAX_LAYERS - 1; i >= 0; i--)
+    {
+        // Render 2D quads for this layer
+
+        RenderQuads(buffer, i);
+
+        // Render text for this layer
+
+        RenderText(buffer, i);
+    }
```
[the remark this answers is not in the document]
Comment (Raelr) [reply engine/render/renderer/renderer/Renderer2D.cpp]: Just made some changes to set it to positive z = closer to the camera.

File: engine/render/renderer/renderer/Renderer2D.h [hunk 3]
```diff
@@ -10,65 +10,121 @@
...
 
-    static void Initialise();
-    static void DestroyRenderer2D();
+    struct FontData
+    {
+        Mat4 transform {};
+        Vec4 uvData {};
+        FColour colour {};
+        Vec4 position {};
+    };
```
[the remark this answers is not in the document]
Comment (Raelr) [reply engine/render/renderer/renderer/Renderer2D.h]: Changed to be more consistent.

File: engine/render/renderer/renderer/Renderer2D.h [hunk 4]
```diff
@@ -10,65 +10,121 @@
...
 {
 class Renderer2D
 {
 public:
 
-    struct GlobalData
+    Renderer2D() = default;
+    void Initialise(const char* const globalDataName);
+    void DrawQuad(const Vec2 position,
+                  const Vec2 scale = Vec2::Zero,
```
[the remark this answers is not in the document]
Comment (Raelr) [reply engine/render/renderer/renderer/Renderer2D.h]: Done!

File: Makefile [hunk 5]
```diff
@@ -7,10 +7,13 @@
 include make/Functions.mk
 include make/Platform.mk
 
+optimisation := -O3
+
 # Set debugging build flags
 DEBUG ?= 1
 ifeq ($(DEBUG), 1)
     override CXXFLAGS += -g -DDEBUG -DCC_LOG_LEVEL=2
+    optimisation :=
```
[the remark this answers is not in the document]
Comment (Raelr) [reply Makefile:16]: Done.

**Pull request merged**